features/yield/FarmingCalculator.tsx

```diff
@@ -0,0 +1,172 @@
+import { Box, TextField, Typography, Grid } from "@material-ui/core";
+
+import { useState, useEffect } from "react";
+import styled from "styled-components";
+import EmpState from "../../containers/EmpState";
+import Balancer from "../../containers/Balancer";
+
+import { getUmaPrice } from "./UmaTokenPrice";
+
+const FormInput = styled.div`
+  margin-top: 20px;
+  margin-bottom: 20px;
+`;
+
+const WEEKS_PER_YEAR = 52;
+
+const YieldCalculator = () => {
+  const { empState } = EmpState.useContainer();
+  const { usdPrice, pool } = Balancer.useContainer();
+  const [umaPrice, setUmaPrice] = useState<number | null>(4.0);
+  const [yUSDAdded, setyUSDAdded] = useState<string | null>("1000");
+  const [USDCAdded, setUSDCAdded] = useState<string | null>("1000");
+  const [fracLiquidity, setFracLiquidity] = useState<string | null>(null);
+  const [umaYieldAmount, setUmaYieldAmount] = useState<string | null>(null);
+  const [usdYieldAmount, setUsdYieldAmount] = useState<string | null>(null);
+  const [apr, setApr] = useState<string | null>(null);
+
+  getUmaPrice().then((price) => {
+    setUmaPrice(price);
+  });
+  //   console.log("PRICE", umaPrice);
+
+  const calculateYield = () => {
+    if (!yUSDAdded || Number(yUSDAdded) < 0) {
+      return null;
+    }
+    if (!USDCAdded || Number(USDCAdded) < 0) {
+      return null;
+    }
+    if (!usdPrice || Number(usdPrice) < 0) {
+      return null;
+    }
+    if (!pool) {
+      return null;
+    }
+    if (!umaPrice) {
+      return null;
+    }
+    const totalValueAddedToPool =
+      Number(USDCAdded) + Number(yUSDAdded) * Number(usdPrice);
+    if (!totalValueAddedToPool || totalValueAddedToPool == 0) {
```

Comment:
I think we should keep all these null or otherwise invalid data checks as close to the top of the function as possible for readability. If anything, I'd recommend combining them all into one if statement

Reply:
This i'll leave for a separate PR. good to do all these reactors in one go